docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+Should it be?-->
 
-Finally, `kubewarden-controller` will create the `policy-server`
-Deployment and a Kubernetes ClusterIP Service to expose it inside of
-the cluster network.
+Finally, `kubewarden-controller` creates the `policy-server` Deployment
+and a Kubernetes ClusterIP Service
+to expose it inside the cluster network.
 
 ### Defining the first policy
 
-This chart shows what happens when the first policy bounded to the default `policy-server` is defined inside of the
-cluster:
+<!--TODO:First mention of 'bound'. I'd like to define what binding or to
+be bound is, how it happens, where, etc?-->
+
+This diagram shows what happens when defining the first policy
+bound to the default `policy-server` in the cluster:
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_02.png)
 
-`kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and,
-as a result of that, it finds the bounded `PolicyServer` and reconciles it.
+<!--TODO:There is ClusterAdmissionPolicy inside and outside the API
+Server box.  What is this signifying? And for PolicyServer? Definition
+of reconcile? Just to find the PolicyServer that matches the
+ClusterAdmissionPolicy?-->
+
+The `kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and
+so finds the bound `PolicyServer` and reconciles it.
 
 ### Reconciliation of `policy-server`
 
-When a `ClusterAdmissionPolicy` is created, modified or deleted a reconciliation loop for the `PolicyServer`
-that owns the policy is triggered inside the `kubewarden-controller`.
-In this reconciliation loop, a ConfigMap with all the polices bounded to 
-the `PolicyServer` is created. Then the a Deployment rollout of the
-interested `policy-server` is started. As a result of that, the new `policy-server`
-instance will be started with the updated configuration.
+<!--TODO:'reconciliation loop', can we show this loop in the diagram
+somehow? ConfigMap, should be in the diagram? In the OCI registry?-->
 
-At start time, `policy-server` reads its configuration and downloads
-all the Kubewarden policies. Policies can be downloaded from remote
-endpoints like HTTP(s) servers and container registries.
+When creating, modifying or deleting a `ClusterAdmissionPolicy`,
+a reconciliation loop activates in `kubewarden-controller`,
+for the `PolicyServer` owning the policy.
+This reconciliation loop creates a ConfigMap with all the polices bound to the `PolicyServer`.
+Then the Deployment rollout of the `policy-server` starts.
+It results in starting the new `policy-server` instance with the updated configuration.
 
-Policies' behaviour can be tuned by users via policy-specific configuration
-parameters. Once all the policies are downloaded, `policy-server` will ensure
-the policy settings provided by the user are valid.
+<!--TODO:Next paragraph. Download all the policies, that the configuration
+specifies, right?-->
```

Comment:
Yes. All the policy servers have a configmap with all the policies it must run with they configuration and information required to run it. Including from where they should be downloaded.

Reply:
Thanks.